src/components/organisms/UserProfileModal/UserProfileModal.tsx

```diff
@@ -136,30 +135,36 @@ export const UserProfileModal: React.FC<UserProfileModalProps> = ({
   );
 };
 
+/**
+ * @debt I believe this relates to Playa features, which are legacy code that will be removed soon
+ * @deprecated legacy tech debt related to Playa, soon to be removed
+ */
 const SuspectedLocation: React.FC<{
   user: WithId<User>;
   currentVenue: WithId<AnyVenue>;
 }> = ({ user, currentVenue }) => {
-  // @debt This will currently load all venues in firebase into memory.. not very efficient
-  useFirestoreConnect("venues");
-  const allVenues = useSelector(orderedVenuesSelector);
+  const { relatedVenues } = useRelatedVenues({
+    currentVenueId: currentVenue.id,
+  });
+
+  const { userLocation } = useWorldUserLocation(user.id);
+  const { lastSeenIn } = userLocation ?? {};
 
   const suspectedLocation = useMemo(
     () => ({
-      venue: allVenues?.find(
-        (v) =>
-          (user.lastSeenIn && user.lastSeenIn[currentVenue?.name ?? ""]) ||
-          v.name === user.room
+      venue: relatedVenues.find(
+        (venue) => lastSeenIn?.[currentVenue.name] || venue.name === user.room
       ),
-      room: allVenues?.find(
-        (v) =>
-          isVenueWithRooms(v) && v.rooms?.find((r) => r.title === user.room)
+      room: relatedVenues.find(
+        (venue) =>
+          isVenueWithRooms(venue) &&
+          venue.rooms?.find((r) => r.title === user.room)
       ),
     }),
-    [user, allVenues, currentVenue]
+    [relatedVenues, lastSeenIn, currentVenue.name, user.room]
   );
 
-  if (!user.room || !allVenues) {
+  if (!user.room || relatedVenues.length === 0) {
```

Comment:
You can extract this second check into a variable `hasRelatedVenues` or something like that.